modules/gojira-core/src/main/java/com/flipkart/gojira/core/injectors/GuiceInjector.java

```diff
@@ -50,6 +51,23 @@ public static void assignInjector(Injector givenInjector) {
     }
   }
 
+  /**
+   * Mark assigned injector as null. Added for un-assigning when running tests.
+   */
+  @VisibleForTesting
+  public static void unAssignInjector() {
```

Comment:
Can we define a graceful shutdowm mechanism for Gojira and make `unAssignInjector()` package protected?

Reply:
We have a mechanism for graceful shutdown via Managed. We are already shutting down connections as part of it. I have added TestQueuedSender shutdown as well. Anything other than these may be identified and added as part of another PR.

Reply:
Can we call the `unAssignInjector()` in Managed and call `Managed.shutdown()` in test cases? The main concern here is the public access to `unAssignInjector()`.

Reply:
I don't think "un-assigning injectors" is the right/correct way to do it and is not a standard practice AFAIK. @VisibleForTesting annotation is added here to capture the reason why it is added and should be fine IMO. It should not be part of Managed.shutdown().